### PR TITLE
Adding kubelet-device-plugin for 1.23/1.24 and time-slicing for 1.25

### PR DESCRIPTION
**Pull request by KCSesh**
**Description of changes:**

* Adds new settings for `kubelet-device-plugin.*` to 1.23/1.24/1.25
* Adds time-slicing content to `kubelet-device-plugin.*` 1.25
    

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: data/settings/1.25.x/kubelet-device-plugin.toml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    [
+        "[NVIDIA K8 Device Plugin](https://github.com/NVIDIA/k8s-device-plugin?tab=readme-ov-file#configuration-option-details)",
+    ],
+]
+tags = ["nvidia", "nvidia-time-slicing"]
+
+[[docs.ref.nvidia-time-slicing-rename-by-default]]
+name_override = "nvidia.time-slicing.rename-by-default"
+description = """
+Specifies the Kubernetes advertised resource as <resource-name>.shared instead of <resource-name>.
```
Comment (piyush-jena) [inline data/settings/1.25.x/kubelet-device-plugin.toml:124]: Something wrong in rendering of this line. Try using `

File: data/settings/1.25.x/kubelet-device-plugin.toml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Therefore, it is recommended to configure time-slicing settings before deploying pods to ensure consistency across all GPU workloads.
+
+<h5>Use Cases</h5>
+
+The time-slicing feature is disabled by default in Bottlerocket. This feature does not provide memory or fault isolation between replicas, and has unique resource request behavior as described [here](https://github.com/NVIDIA/k8s-device-plugin?tab=readme-ov-file#with-cuda-time-slicing). 
+According to [NVIDIA](https://developer.nvidia.com/blog/improving-gpu-utilization-in-kubernetes/#virtualization_with_vgpu), this feature is best used for over subscribing the GPU when needing to run multiple applications that are not latency-sensitive or can tolerate jitter.
+
+<h5>Example Usage</h5>
+
+In a Bottlerocket Kubernetes NVIDIA variant, if the below configuration were applied to a node with 8 GPUs on it, the plugin would now advertise 80 nvidia.com/gpu.shared resources to Kubernetes instead of 8 (8 GPU’s x 10 replicas = 80).
```
Comment (piyush-jena) [inline data/settings/1.25.x/kubelet-device-plugin.toml:28]: Surround nvidia.com/gpu.shared with `

File: data/settings/1.25.x/kubelet-device-plugin.toml [hunk 3]
```diff
@@ -0,0 +1,152 @@
+[[docs.tag.nvidia]]
+heading = "Settings related to k8 nvidia device plugin"
+description = """
+See the [nvidia-k8s-device-plugin](https://github.com/NVIDIA/k8s-device-plugin?tab=readme-ov-file#with-cuda-time-slicing) for more details.
+"""
+
+[[docs.tag.nvidia-time-slicing]]
```
Comment (arnaldo2792) [inline data/settings/1.25.x/kubelet-device-plugin.toml:7]: This should be its own section like `Bootstrap Containers`, and a simpler description should be provided in the API documentation.

**Review by piyush-jena — APPROVED** [review]
1. Checked nvidia settings api documentation in v1.23, v1.24, v1.25
2. Checked timeslicing documentation in v1.25
3. Verified some of the driver versions and browsed some of the settings, concepts
LGTM!

**Review by arnaldo2792 — APPROVED** [review]

**Review by ytsssun — APPROVED** [review]

**Pull request merged**